infrastructure/resource_group_level/modules/dtfs-central-api.bicep

```diff
@@ -0,0 +1,169 @@
+var connectionStrings = {
+  EXTERNAL_API_URL: {
+    type: 'Custom'
+    // TODO:FN-421 get external-api (reference-data-proxy) default hostname.
+    value: 'https://tfs-feature-external-api.azurewebsites.net/'
+  }
+  MONGO_INITDB_DATABASE: {
+    type: 'Custom'
+    value: cosmosDbDatabaseName
+  }
+  MONGODB_URI: {
+    type: 'Custom'
+    value: mongoDbConnectionString
+  }
+}
+
+resource cosmosDbAccount 'Microsoft.DocumentDB/databaseAccounts@2023-04-15' existing = {
+  name: cosmosDbAccountName
+}
+
+resource dtfsCentralApi 'Microsoft.Web/sites@2022-09-01' = {
+  name: dtfsCentralApiName
+  location: location
+  tags: {
+    Environment: 'Preproduction'
+  }
+  kind: 'app,linux,container'
+  properties: {
+    httpsOnly: false
+    serverFarmId: appServicePlanId
+    siteConfig: {
+      numberOfWorkers: 1
+      linuxFxVersion: 'DOCKER|${dockerImageName}'
+      acrUseManagedIdentityCreds: false
+      alwaysOn: true
+      http20Enabled: true
+      functionAppScaleLimit: 0
+      // non-default parameter
+      logsDirectorySizeLimit: 100 // default is 35
+      // The following Fields have been added after comparing with dev
+      vnetRouteAllEnabled: true
+      ftpsState: 'Disabled'
+      scmMinTlsVersion: '1.0'
+      remoteDebuggingVersion: 'VS2019'
+      httpLoggingEnabled: true // false in staging, true in prod
+    }    
+    virtualNetworkSubnetId: appServicePlanEgressSubnetId
+  }
+}
+
+resource dtfsCentralApiSettings 'Microsoft.Web/sites/config@2022-09-01' = {
+  parent: dtfsCentralApi
+  name: 'appsettings'
+  properties: appSettings
+}
+
+resource dtfsCentralApiConnectionStrings 'Microsoft.Web/sites/config@2022-09-01' = {
+  parent: dtfsCentralApi
+  name: 'connectionstrings'
+  properties: connectionStrings
+}
+
+// The private endpoint is taken from the cosmosdb/private-endpoint export
+resource dtfsCentralApiPrivateEndpoint 'Microsoft.Network/privateEndpoints@2022-11-01' = {
+  name: privateEndpointName
+  location: location
+  tags: {
+    Environment: 'Preproduction'
+  }
+  properties: {
+    privateLinkServiceConnections: [
+      {
+        name: privateEndpointName
+        properties: {
+          privateLinkServiceId: dtfsCentralApi.id
+          groupIds: [
+            'sites'
+          ]
+          privateLinkServiceConnectionState: {
+            status: 'Approved'
+            actionsRequired: 'None'
+          }
+        }
+      }
+    ]
+    manualPrivateLinkServiceConnections: []
+    subnet: {
+      id: privateEndpointsSubnetId
+    }
+    ipConfigurations: []
+    // Note that the customDnsConfigs array gets created automatically and doesn't need setting here.
+  }
+}
+
+resource applicationInsights 'Microsoft.Insights/components@2020-02-02' = {
+  name: applicationInsightsName
+  location: location
+  tags: {
+    Environment: 'Preproduction'
+  }
+  kind: 'web'
+  properties: {
+    Application_Type: 'web'
+    Flow_Type: 'Redfield'
+    RetentionInDays: 90
```

Comment:
Note that Dev has `Request_Source: 'IbizaAIExtensionEnablementBlade'`
I haven't put it in and was wondering if it would appear automatically, but it hasn't. I don't understand what it is for. Shall I just add it anyway?

Reply:
Yeah, I got almost nothing from Google, though I do see sporadic mentions here and there. I suspect it was a temporary thing during development. Ibiza seems to be some sort of codename.

Similarly, the Flowtype of `Redfield` appears non-standard.
I'll stick with the plan of keeping them for now, and simplifying when I have a working environment.